pkg/cloud/external_storage.go

```diff
@@ -184,42 +200,50 @@ type ExternalStorageConstructor func(
 	context.Context, ExternalStorageContext, cloudpb.ExternalStorage,
 ) (ExternalStorage, error)
 
-// NewExternalStorageAccessor creates an uninitialized ExternalStorageAccessor.
-func NewExternalStorageAccessor() *ExternalStorageAccessor {
-	return &ExternalStorageAccessor{ready: make(chan struct{})}
+type EarlyBootExternalStorageConstructor func(
```

Comment:
Hrm. I'm all for better names as I dislike many of the names in this package including this one.

But, the reason why we have the two different names here is that typically "Factory" refers to a method that knows how to select between multiple implementations of a given interface for some configuration. 

For example, if you have an Interface A with implementations _f_ and _g_ an AFactory would be what is responsible for choosing between _f_ and _g_.  

Here the FromURIFactory is a factory because it knows how to take a URI and then route that to a specific ExternalStorage implementation.  The `Constructor` types is implemented by an ExternalStorage implementation and is what is _called by a factory_.

Reply:
ah, thank you for the explanation. This makes sense. I had not realized Factory had a specific meaning.